HotorNot/app/src/main/java/com/example/hotornot/fragments/RegistrationScreenFragment.kt

```diff
@@ -1,35 +1,175 @@
-import com.example.hotornot.R
+import android.widget.Toast
+import androidx.navigation.fragment.findNavController
+import com.example.hotornot.*
+import com.example.hotornot.data.User
+import com.example.hotornot.enums.Gender
 import com.example.hotornot.databinding.FragmentRegistrationScreenBinding
-import com.example.hotornot.databinding.FragmentSplashScreenBinding
-import kotlinx.android.synthetic.main.fragment_registration_screen.*
 
 class RegistrationScreenFragment : Fragment() {
 
     private lateinit var binding: FragmentRegistrationScreenBinding
+    private lateinit var preferencesUtil: PreferencesUtil
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?,
     ): View {
         binding = FragmentRegistrationScreenBinding.inflate(layoutInflater, container, false)
-        showInterests()
+        binding.imgReg.setImageResource(R.drawable.ic_friends_logo)
         return binding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        preferencesUtil = PreferencesUtil.getInstance(view.context)
+        onFirstNameTextChangeListener()
+        onLastNameTextChangeListener()
+        onEmailTextChangeListener()
+        showInterests()
+        isRegisterUser()
+    }
+
+    private fun onFirstNameTextChangeListener() {
+        binding.txtEmail.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {}
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                binding.txtFirstName.error = isEmptyFirstName()
+                binding.btnReg.isEnabled = true
+            }
+        })
+    }
+
+    private fun onLastNameTextChangeListener() {
+        binding.lastName.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {}
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                binding.txtLastName.error = isEmptyLastName()
+                binding.btnReg.isEnabled = true
+            }
+        })
+    }
+
+    private fun onEmailTextChangeListener() {
+        binding.txtEmail.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {}
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                binding.txtEmailReg.error = isValidEmail()
+                binding.btnReg.isEnabled = true
+            }
+        })
+    }
+
+    private fun isEmptyFirstName(): String? {
```

Comment:
When the method have return value is better to say "getAnything" or just use get prefix.When return boolean value ,we can have questionable name.